projects/core/src/cms/facade/page-meta.service.ts

```diff
@@ -40,14 +39,19 @@ export class PageMetaService {
     robots: 'resolveRobots',
   };
 
-  getMeta(): Observable<PageMeta | null> {
-    return this.cms.getCurrentPage().pipe(
+  protected meta$: Observable<PageMeta | null> = defer(() =>
+    this.cms.getCurrentPage().pipe(
       filter(Boolean),
       switchMap((page: Page) => this.getMetaResolver(page)),
       switchMap((metaResolver: PageMetaResolver) =>
         metaResolver ? this.resolve(metaResolver) : of(null)
-      )
-    );
+      ),
+      shareReplay({ bufferSize: 1, refCount: true })
+    )
+  );
```

Comment:
Each subscriber of `defer` invokes the inner function independently and gets his own instance of the returned observable. By putting `shareReplay` inside the function, we would loose the benefit of sharing the final results in between all subcribers.

Instead we should wrap only the invocation of  `this.cms.getCurrentPage()` inside `defer`. Thanks to that, calling `this.cms.getCurrentPage()` will happen only when someone subscribes to the `meta$` observable. At the same time we pipe `shareReplay` in the end of the operators chain, so the final value of the `meta$` observable will be shared among all subscribers:

```ts
defer(() => this.cms.getCurrentPage()).pipe(
  filter,
  switchMap,
  switchMap,
  shareReplay,
);
```

Reply:
thanks a lot for this, good stuff!